src/full/Agda/Syntax/Internal.hs

```diff
@@ -757,6 +757,11 @@ instance Null (Substitution' a) where
   null IdS = True
   null _   = False
 
+-- | Wrapper for types that do not contain variables (so applying a substitution is the identity).
+--   Useful if you have a structure of types that support substitution mixed with types that don't
+--   and need to apply a substitution to the full structure.
+newtype NoSubst t a = NoSubst { unNoSubst :: a }
```

Comment:
Isn't this usually called `Closed`?  But maybe "closed" is again ambiguous, and `NoSubst` is pretty unambiguous.

Reply:
As a disclaimer, I didn't implement this type, I just moved it here because it didn't really belonged in the Mimer-specific part of the Agda codebase imo. But I'm happy to discuss alternative names for it, I don't have a strong preference for NoSubst vs Closed.